src/github_changelog/conventional.clj

```diff
@@ -38,15 +38,39 @@
 (defn parse-issues [config pull]
   (apply concat ((juxt jira-issues github-issues) config pull)))
 
+(defn parse-revert [{:keys [user repo]} {:keys [title body]}]
+  (if (starts-with? title "Revert ")
+    (let [revert-prefix (format "Reverts %s/%s#" user repo)
+          [prefix pull-id] (map join (split-at (count revert-prefix) body))]
+      (if (starts-with? prefix revert-prefix)
+        (parse-int pull-id)))))
+
 (defn parse-pull [config {:keys [title] :as pull}]
-  (if-let [[_ type scope subject] (re-find header-pattern title)]
-    {:type type
-     :scope scope
-     :subject subject
-     :pull-request pull
-     :issues (parse-issues config pull)}))
+  (if-let [pull-id (parse-revert config pull)]
+    {:revert-pull pull-id
+     :pull-request pull}
+    (if-let [[_ type scope subject] (re-find angular-pattern title)]
+      {:type type
+       :scope scope
+       :subject subject
+       :pull-request pull
+       :issues (parse-issues config pull)})))
+
+(defn reverted-ids [pulls]
+  (->> (map :revert-pull (seq pulls))
+       (remove nil?)
+       (set)))
+
+(defn filter-reverted [pulls {:keys [revert-pull pull-request] :as pull}]
+  (let [reverted-pulls (reverted-ids pulls)
```

Comment:
You are realizing this set for every element of the seq, when you could take the whole sequence of pull requests and do it once.